server/src/main/java/org/elasticsearch/common/geo/parsers/GeoWKTParser.java

```diff
@@ -228,9 +238,9 @@ private static GeometryCollectionBuilder parseGeometryCollection(StreamTokenizer
         if (nextEmptyOrOpen(stream).equals(EMPTY)) {
             return null;
         }
-        GeometryCollectionBuilder builder = new GeometryCollectionBuilder().shape(parseGeometry(stream));
+        GeometryCollectionBuilder builder = new GeometryCollectionBuilder().shape(parseGeometry(stream, null));
```

Comment:
nit: maybe you can use GeoShapeType.GEOMETRYCOLLECTION here instead of null and then check for that enum in the method where the exception might be thrown?